server/src/models/mod.rs

```diff
@@ -61,20 +71,109 @@ impl ErrorBound {
     }
 }
 
-/// Enable equal and not equal for `ErrorBound` and `f32`.
+/// Enable equal and not equal for [`ErrorBound`] and [`f32`].
 impl PartialEq<ErrorBound> for f32 {
     fn eq(&self, other: &ErrorBound) -> bool {
         self.eq(&other.0)
     }
 }
 
-/// Enable less than and greater than for `ErrorBound` and `f32`.
+/// Enable less than and greater than for [`ErrorBound`] and [`f32`].
 impl PartialOrd<ErrorBound> for f32 {
     fn partial_cmp(&self, other: &ErrorBound) -> Option<Ordering> {
         self.partial_cmp(&other.0)
     }
 }
 
+/// Model that uses the fewest number of bytes per value.
+pub struct SelectedModel {
+    /// Id of the model type that created this model.
+    pub model_type_id: u8,
+    /// Index of the last data point in the `UncompressedSegment` that the
+    /// selected model represents.
+    pub end_index: usize,
+    /// The selected model's minimum value.
+    pub min_value: Value,
+    /// The selected model's maximum value.
+    pub max_value: Value,
+    /// Data required in addition to `min` and `max` for the model to
+    /// reconstruct the values it represents when given a specific timestamp.
+    pub values: Vec<u8>,
+}
+
+impl SelectedModel {
+    fn new(
+        model_type_id: u8,
+        end_index: usize,
+        min_value: Value,
+        max_value: Value,
+        values: Vec<u8>,
+    ) -> Self {
+        Self {
+            model_type_id,
+            end_index,
+            min_value,
+            max_value,
+            values,
+        }
+    }
+}
+
+/// Select the model that uses the fewest number of bytes per value.
+pub fn select_model(
```

Comment:
Consider moving the functionality of this function to the `SelectedModel::new()` method.

Reply:
Fixed.